From b83322655b6bbf616fb8021a6a69584ae1486f7e Mon Sep 17 00:00:00 2001
From: Dan Williams <dcbw@redhat.com>
Date: Sat, 1 Jan 2011 18:00:50 -0600
Subject: [PATCH 6/8] modem: handle IP Method changes after modem has been created

Some modems don't know their IP method when they are created, but
will update it later.  Handle the changed signal for that.
---
 src/modem-manager/nm-modem.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/src/modem-manager/nm-modem.c b/src/modem-manager/nm-modem.c
index e07b818..b3f7eaa 100644
--- a/src/modem-manager/nm-modem.c
+++ b/src/modem-manager/nm-modem.c
@@ -876,6 +876,12 @@ modem_properties_changed (DBusGProxy *proxy,
 		priv->mm_enabled = g_value_get_boolean (value);
 		g_object_notify (G_OBJECT (self), NM_MODEM_ENABLED);
 	}
+
+	value = g_hash_table_lookup (props, "IpMethod");
+	if (value && G_VALUE_HOLDS_UINT (value)) {
+		priv->ip_method = g_value_get_uint (value);
+		g_object_notify (G_OBJECT (self), NM_MODEM_IP_METHOD);
+	}
 }
 
 /*****************************************************************************/
-- 
1.7.3.4

